src/FSharp.Analyzers/PartialAppAnalyzer.fs

```diff
@@ -24,240 +32,288 @@ let collectExprFromMatchClauses (clauses : SynMatchClause list) =
     )
     |> List.concat
 
-let rec visitApp (handler : Handler) (depth : int) (expr : SynExpr) =
+let rec (|AppOfIdent|_|) (expr : SynExpr) =
+    match expr with
+    | SynExpr.App (funcExpr = SynExpr.Ident ident) -> Some ident
+    | SynExpr.App (funcExpr = SynExpr.App _ as a) ->
+        match a with
+        | AppOfIdent i -> Some i
+        | _ -> None
+    | _ -> None
+
+let rec (|AppOfLongIdent|_|) (expr : SynExpr) =
+    match expr with
+    | SynExpr.App (funcExpr = SynExpr.LongIdent (longDotId = ident)) -> Some ident
+    | SynExpr.App (funcExpr = SynExpr.App _ as a) ->
+        match a with
+        | AppOfLongIdent i -> Some i
+        | _ -> None
+    | _ -> None
+
+let collectPipedArgs (pipeHandler : PipeHandler) (funcExpr : SynExpr) (argExpr : SynExpr) =
+    match funcExpr with
+    | SynExpr.App (funcExpr = SynExpr.LongIdent (longDotId = longDotId) ; argExpr = argExpr2) ->
+        longDotId.LongIdent
+        |> Seq.tryHead
+        |> Option.iter (fun i ->
+            let pipedArgs, target =
+                match i.idText with
+                | "op_PipeRight" -> 1, Some argExpr
+                | "op_PipeRight2" -> 2, Some argExpr
+                | "op_PipeRight3" -> 3, Some argExpr
+                // not needed for op_PipeLeft
+                | "op_PipeLeft2" -> 2, Some argExpr2
+                | "op_PipeLeft3" -> 3, Some argExpr2
+                | _ -> 0, None
+
+            if pipedArgs > 0 then
+                target
+                |> Option.iter (
+                    function
+                    | AppOfIdent i -> pipeHandler (i.idRange, pipedArgs)
+                    | AppOfLongIdent i -> pipeHandler (i.Range, pipedArgs)
+                    | _ -> ()
+                )
+        )
+    | _ -> ()
+
+let rec visitApp (handlers : Handlers) (depth : int) (expr : SynExpr) =
```

Comment:
Maybe something to revisit in the future but this list doesn't seem complete.
`MatchBang`, `TryWith`, `TryFinally` are missing.